File: implementation/scipamato/scipamato-core-web/src/main/java/ch/difty/scipamato/web/model/CodeClassModel.java
package ch.difty.scipamato.web.model;

import ch.difty.scipamato.entity.CodeClass;
import ch.difty.scipamato.persistence.CodeClassService;

/**
 * Model that offers a wicket page to load {@link CodeClass}es.
 *
 * @author u.joss
 */
public class CodeClassModel extends CodeClassLikeModel<CodeClass, CodeClassService> {

    private static final long serialVersionUID = 1L;

    public CodeClassModel(final String languageCode) {
        super(languageCode);
    }

}

File: public/public-web/src/main/java/ch/difty/scipamato/publ/web/model/CodeClassModel.kt
package ch.difty.scipamato.publ.web.model

import ch.difty.scipamato.common.web.model.CodeClassLikeModel
import ch.difty.scipamato.publ.entity.CodeClass
import ch.difty.scipamato.publ.persistence.api.CodeClassService

/**
 * Model that offers a wicket page to load [CodeClass]es.
 *
 * @author u.joss
 */
class CodeClassModel(languageCode: String) : CodeClassLikeModel<CodeClass, CodeClassService>(languageCode) {

    /** just delegating to super, but making load visible to test  */
    @Suppress("RedundantOverride")
    override fun load(): List<CodeClass> = super.load()

    companion object {
        private const val serialVersionUID = 1L
    }
}
